packages/amplify_datastore/example/integration_test/query_test/query_predicate_test/string_query_predicate_test.dart

```diff
@@ -0,0 +1,276 @@
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify.dart';
+
+import '../../utils/query_predicate_utils.dart';
+import '../../utils/setup_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('type String', () {
+    // models used for all tests
+    var models = [
+      StringTypeModel(value: 'foo'),
+      StringTypeModel(value: 'bar'),
+      StringTypeModel(value: 'foo'),
+      StringTypeModel(value: 'bar'),
+      StringTypeModel(value: 'abcd'),
+      StringTypeModel(value: 'abce'),
+      StringTypeModel(value: 'abcf'),
+      StringTypeModel(value: ''),
+      StringTypeModel(),
+    ];
+
+    // non-null models used for all tests
+    var nonNullModels = models.where((e) => e.value != null).toList();
+
+    // distinct list of values in the test models
+    var values = models.map((e) => e.value).toSet().toList();
+
+    // distinct list of non-null values in the test models
+    var nonNullValues = values.whereType<String>().toList();
+
+    setUpAll(() async {
+      await configureDataStore();
+      await clearDataStore();
+      for (var model in models) {
+        await Amplify.DataStore.save(model);
+      }
+    });
+    testWidgets('eq()', (WidgetTester tester) async {
+      // test against all values
+      for (var value in values) {
+        var expectedModels =
+            models.where((model) => model.value == value).toList();
+        await testStringQueryPredicate(
+          queryPredicate: StringTypeModel.VALUE.eq(value),
+          expectedModels: expectedModels,
+        );
+      }
+
+      // test with no match
+      await testStringQueryPredicate(
+        queryPredicate: StringTypeModel.VALUE.eq('no match'),
+        expectedModels: [],
+      );
+    });
+
+    testWidgets('ne()', (WidgetTester tester) async {
+      // test against all values
+      for (var value in values) {
+        // update `nonNullModels` to `models` when #823 is fixed
+        // see: https://github.com/aws-amplify/amplify-flutter/issues/823
+        var expectedModels =
+            nonNullModels.where((model) => model.value != value).toList();
+        await testStringQueryPredicate(
+          queryPredicate: StringTypeModel.VALUE.ne(value),
+          expectedModels: expectedModels,
+        );
+      }
+
+      // test with match all models
+      await testStringQueryPredicate(
+        queryPredicate: StringTypeModel.VALUE.ne('no match'),
+        // update `nonNullModels` to `models` when #823 is fixed
+        // see: https://github.com/aws-amplify/amplify-flutter/issues/823
+        expectedModels: nonNullModels,
+      );
+    });
+
+    testWidgets('lt()', (WidgetTester tester) async {
+      // test against all (non-null) values
+      for (var value in nonNullValues) {
+        var expectedModels = nonNullModels
+            .where((model) => model.value!.compareTo(value) < 0)
```

Comment:
Just curious, we are sure about that the implementation of the comparison operators of string in DataStore is the same as the `compareTo` in Dart, correct?

Reply:
If it weren't, the tests would fail 😄 

Serious answer though, there are scenarios where I had to avoid using compareTo, in particular when using nullable types. But in general, yes, it is the same.